cluster/types.go

```diff
@@ -0,0 +1,93 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cluster
+
+// Params defines an intended charon cluster configuration.
+type Params struct {
```

Comment:
@corverroos the public fields make this structure mutable, unfortunatelly, making it less functional. But I understand that to do the json encoding would require a bit of boilerplate code if we made them private, so perhaps it is acceptable?

Reply:
considering we are already writing a MarshallJSON and UnmarshallJSOn for Params, there is no need for these fields to be public?

Reply:
So this is a DTO, ie. a pure value, fields should be public.

Reply:
ok, grouping functions by type.